src/ert/job_queue/driver.py

```diff
@@ -101,69 +105,119 @@ async def kill(self, realization: "RealizationState"):
         self._processes[realization].kill()
         realization.verify_kill()
 
-
+LSF_STATUSES = [
+               "PEND",
+               "SSUSP",
+               "PSUSP",
+               "USUSP",
+               "RUN",
+               "EXIT",
+               "ZOMBI",
+               "DONE",
+               "PDONE",
+               "UNKWN",
+]
 class LSFDriver(Driver):
     def __init__(self, queue_options):
         super().__init__(queue_options)
 
         self._realstate_to_lsfid: Dict["RealizationState", str] = {}
         self._lsfid_to_realstate: Dict[str, "RealizationState"] = {}
+        self._max_attempt = 100
+        self._MAX_ERROR_COUNT = 100
+        self._statuses = {}
         self._submit_processes: Dict[
             "RealizationState", asyncio.subprocess.Process
         ] = {}
-
+        self._SLEEP_PERIOD = 3
         self._currently_polling = False
 
     async def submit(self, realization: "RealizationState"):
         submit_cmd = [
-            "bsub",
+            self.get_option("BSUB_CMD") if self.has_option("BSUB_CMD") else "bsub",
             "-J",
             f"poly_{realization.realization.run_arg.iens}",
             realization.realization.job_script,
             realization.realization.run_arg.runpath,
         ]
         assert shutil.which(submit_cmd[0])  # does not propagate back..
-        process = await asyncio.create_subprocess_exec(
-            *submit_cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        self._submit_processes[realization] = process
-
-        # Wait for submit process to finish:
-        output, error = await process.communicate()
-        print(output)  # FLAKY ALERT, we seem to get empty
-        print(error)
-
-        try:
+        current_attempt = 0
+        while current_attempt < self._max_attempt:
+            current_attempt += 1
+            try:
+                process = await asyncio.create_subprocess_exec(
+                    *submit_cmd,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                self._submit_processes[realization] = process
+
+                # Wait for submit process to finish:
+                output, error = await process.communicate()
+                if (process.returncode != 0):
+                    raise Exception(f"bsub returned non-zero exitcode: {process.returncode}")
```

Comment:
We should try to not use the most general exception, and rather use a specific one. In this situation, I think I would avoid using an exception at all to catch this scenario. Since we are assuming the bsub command to be flaky, we can write it as normal code, and not exceptional code.

Reply:
I changed it, but forgot to move the time.sleep() and continue statement. The tests saved me already 30 minutes after writing them :D